knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -155,27 +174,41 @@ func applyKnativeKafka(instance *operatorv1alpha1.KnativeKafka, api client.Clien
 	return nil
 }
 
-func installKnativeKafkaChannel(apiclient client.Client) error {
-	manifest, err := mfc.NewManifest(kafkaChannelManifestPath(), apiclient, mf.UseLogger(log.WithName("mf")))
+func installKnativeKafkaChannel(instance *operatorv1alpha1.KnativeKafka, apiclient client.Client) error {
+	manifest, err := kafkaChannelManifest(instance, apiclient)
 	if err != nil {
-		return fmt.Errorf("failed to load KafkaChannel manifest: %w", err)
+		return fmt.Errorf("failed to load or transform KafkaChannel manifest: %w", err)
 	}
 
 	log.Info("Installing Knative KafkaChannel")
 	if err := manifest.Apply(); err != nil {
 		return fmt.Errorf("failed to apply KafkaChannel manifest: %w", err)
 	}
-	if err := checkDeployments(&manifest, apiclient); err != nil {
+	if err := checkDeployments(manifest, apiclient); err != nil {
 		return fmt.Errorf("failed to check deployments: %w", err)
 	}
 	log.Info("Knative KafkaChannel installation is ready")
 	return nil
 }
 
+func kafkaChannelManifest(instance *operatorv1alpha1.KnativeKafka, apiClient client.Client) (*mf.Manifest, error) {
+	manifest, err := mfc.NewManifest(kafkaChannelManifestPath(), apiClient, mf.UseLogger(log.WithName("mf")))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load KafkaChannel manifest: %w", err)
+	}
+
+	manifest, err = manifest.Transform(mf.InjectOwner(instance))
```

Comment:
That's why the owner refs are only set for the channel resources, which live in `knative-eventing` namespace (same namespace as the CR).
It is not set for the KafkaSource resources, which live in `knative-sources` namespace.

> Otherwise if I create this in namespace foo, how are the owner refs going to work?

You can't create the KnativeKafka CR in a namespace other than `knative-eventing`. This is similar to existing CRs for serving and eventing. That check is done in another PR (https://github.com/openshift-knative/serverless-operator/pull/510/commits/d2895c8729793c98c9bfcd78606ea1f3d6b1865a#diff-186644545bff629e539219c7f924fc96R111) already but there's a lot of stuff to merge until that check is enabled.

Reply:
Hmm, ok.

Reply:
@matzew has some thoughts about using `knative-eventing` for KafkaSource as well, but I don't want to do this until I merge my existing work.

Also, not sure if it requires any changes in the upstream code.

Reply:
@aliok let's track it

Reply:
@matzew already added in the umbrella ticket: https://github.com/openshift-knative/serverless-operator/issues/486